### PR TITLE
Remove unnecessary symbol quote

### PR DESCRIPTION
**Pull request by carolinemengbq**
Related to issue amzn#87

Removing an unnecessary parameter SymbolVariant in the WriteSymbolText method. Should always be using GetSymbolVariant to determine if the symbol or field name needs to be written with single quotes. The changes are matched to ion-Java, and the vector tests will also go through the WriteSymbolText method - all the existing tests pass.

Unit tests added with sample cases of quotation and no quotations.

### CODE REVIEW

File: Amazon.IonDotnet.Tests/Internals/TextWriterTest.cs [hunk 1]
```diff
@@ -0,0 +1,87 @@
...
+                SystemSymbols.IonSymbolTable + "::" +
+                "{" +
+                symbol +
+                "}\n" +
+                "$10\n" +
+                "$11\n";
+            var ionValueFactory = new ValueFactory();
+            var datagram = IonLoader.Default.Load(text);
+            datagram.Add(ionValueFactory.NewSymbol(newSym));
+            datagram.Add(ionValueFactory.NewSymbol(new SymbolToken(null, 13)));
```
Comment (tgregg) [inline Amazon.IonDotnet.Tests/Internals/TextWriterTest.cs:75]: This should actually fail on write because $13 is out of range of the symbol table. The fix for that is out of scope for this PR, so please just remove references to $13 from this test and create a new issue to raise an error if the user attempts to write an out-of-range symbol with unknown text.

File: Amazon.IonDotnet.Tests/Internals/TextWriterTest.cs [hunk 2]
```diff
@@ -0,0 +1,87 @@
...
+
+
+        [TestMethod]
+        [DataRow("+inf", "nan", "s1 '+inf' 'nan' $13")]
+        [DataRow("s2", "abc", "s1 s2 abc $13")]
+        public void WriteSymbolWithSymbolTable(String tableSym, String newSym, String expectedText)
+        {
+            var symbol = "symbols:[\"s1\", \"" + tableSym + "\"]";
+
+                var text =
```
Comment (tgregg) [inline Amazon.IonDotnet.Tests/Internals/TextWriterTest.cs:65]: Please fix the indentation.

File: Amazon.IonDotnet.Tests/Internals/TextWriterTest.cs (hunk 1 above)
Comment (carolinemengbq) [reply Amazon.IonDotnet.Tests/Internals/TextWriterTest.cs]: Done.

File: Amazon.IonDotnet.Tests/Internals/TextWriterTest.cs (hunk 2 above)
Comment (carolinemengbq) [reply Amazon.IonDotnet.Tests/Internals/TextWriterTest.cs]: Done.
